tests/Platforms/MySQLPlatformTest.php

```diff
@@ -20,4 +21,10 @@ public function testHasCorrectDefaultTransactionIsolationLevel(): void
             $this->platform->getDefaultTransactionIsolationLevel()
         );
     }
+
+    public function testDropIndexSQLRequiresTable(): void
+    {
+        $this->expectException(InvalidArgumentException::class);
+        $this->platform->getDropIndexSQL('foo');
+    }
```

Comment:
Picking MySQL looks arbitrary. Why do we not have a unit test for `AbstractPlatform`? Also, this looks like it should be in a separate commit.

Reply:
> Picking MySQL looks arbitrary.

The logic of requiring the optional `$table` at runtime already exists in the MySQL platform but it wasn't covered with a test. I added it for symmetry.

> Why do we not have a unit test for `AbstractPlatform`?

Because the semantics of this method is inconsistent across the platforms: most of the platforms do not use `$table` since the index names there are scoped to the schema but MySQL and SQL Server use and require it because their index names are scoped the table to which the index applies.

A more consistent approach might be to make `$table` non-optional at the `AbstractPlatform` level but then it would require passing it while in most cases it wouldn't be used. So I decided not to change the API until/unless we come up with a better idea.

> Also, this looks like it should be in a separate commit.

Good idea. Done.